mhs/common/mhs_common/messages/ebxml_nack_envelope.py

```diff
@@ -0,0 +1,40 @@
+"""This module defines the envelope used to wrap the negative acknowledgement messages to be sent to a remote MHS in
+response to an asynchronous request."""
+
+from __future__ import annotations
+
+import copy
+from typing import Dict
+
+from defusedxml import ElementTree
+
+import mhs_common.messages.ebxml_envelope as ebxml_envelope
+import mhs_common.messages.common_ack_envelope as common_ack_envelope
+
+EBXML_TEMPLATE = "ebxml_nack"
+
+
+class EbxmlNackEnvelope(common_ack_envelope.CommonEbxmlAckEnvelope):
+    """An envelope that contains a negative acknowledgement of an asynchronous request from a remote MHS."""
+
+    def __init__(self, message_dictionary: Dict[str, str]):
+        """Create a new EbxmlNackEnvelope that populates the message with the provided dictionary.
+
+        :param message_dictionary: The dictionary of values to use when populating the template.
+        """
+        message_dictionary = copy.deepcopy(message_dictionary)
+        message_dictionary[ebxml_envelope.ACTION] = 'MessageError'
```

Comment:
TODO-this is hardcoded already in the template (ie can remove the hardocoding in one of these 2 places)